src/optionsProp.ts

```diff
@@ -1,10 +1,25 @@
 import { isNullOrUndefined } from 'util';
 import { DecoratorKeys } from './internal/constants';
-import { globalOptions } from './internal/data';
+import { childs, globalOptions } from './internal/data';
 import { assignMetadata, getName } from './internal/utils';
 import { logger } from './logSettings';
 import { IModelOptions } from './types';
 
+function assignToParentClass(target: any) {
```

Comment:
why is this in the `modelOptions` decorator? when the decorator dosnt get called, then this will never occur, and the decorator is meant to be completely optional